File: Task2/blooddosyst/src/main/java/ua/nure/mossurd/blooddosyst/dto/DonationEventDto.java
package ua.nure.mossurd.blooddosyst.dto;

import ua.nure.mossurd.blooddosyst.entity.DonationEvent;
import ua.nure.mossurd.blooddosyst.enums.DonationEventStatus;

import java.time.LocalDateTime;

public record DonationEventDto(
        Integer id,
        LocalDateTime dateTime,
        String eventAddress,
        DonationEventStatus status,
        String notes
) {
    public DonationEventDto(DonationEvent donationEvent) {
        this(donationEvent.getId(),
                donationEvent.getDateTime(),
                donationEvent.getEventAddress(),
                donationEvent.getEventStatus(),
                donationEvent.getNotes());
    }
}

File: Task4/akpz-pzpi-22-2-mossur-daria-task4/BloodDo/app/src/main/java/ua/nure/mossurd/blooddonation/dto/DonationEventDto.kt
package ua.nure.mossurd.blooddonation.dto

import ua.nure.mossurd.blooddonation.enums.DonationEventStatus
import java.io.Serializable
import java.util.Date

data class DonationEventDto(
    val id: Integer,
    val dateTime: Date,
    val eventAddress: String,
    val status: DonationEventStatus,
    val notes: String
) : Serializable
